src/EnergyPlus/CurveManager.cc

```diff
@@ -2827,23 +2787,23 @@ namespace CurveManager {
                         std::string_view curveFieldText)
     {
         // Returns true if errors found
-        int curveDim = state.dataCurveManager->PerfCurve(CurveIndex).NumDims;
+        auto &thisCurve = state.dataCurveManager->PerfCurve(CurveIndex);
```

Comment:
`validDims` is passed into this function by value. Also, it should probably be a `gsl::span` because most (all?) of the actual arguments to it are essentially `constexpr` and don't require a heap allocation.

Reply:
Another general comment. I am not a fan of setting up and passing a bunch of string parameters to a function for the rare case where it has to print an error. Very non-MCCF. Better to have a separate error function IMO.

Reply:
Yep, agreed.  I'm not sure when to make that sort of change in this PR, now or after some other structural changes.  But anyway, I agree, and hopefully a better pattern arises organically.

Reply:
Interesting.  It is passed as a plain vector because in the client code we call it with a temporary initializer list:

`result = CheckCurveDims(state, myCurveIndex, {0, 1}, "routineName", ....);`

So the intializer list is of course passed to the vector constructor.  I think if I make this a vector reference, it's going to complain that an initializer list is no longer a valid argument type to pass in.  But yes, I think a span object would work here, I'll have to try.